File: July-18/01-07-18/python/01-Pro-and-con-list.py
for _ in range(int(input())):
    n = int(input())
    ar = [0]*n
    sad = 0
    
    for i in range(0, n):
        n1, n2 = map(int, input().split())
        ar[i] = n1 + n2
        sad += n2
    ar.sort()
    ans = ar[n-1] + ar[n-2] - sad
    print(ans)

File: July-18/01-07-18/c++/01-Pro-and-con-list.cpp
#include<bits/stdc++.h>

using namespace std;
typedef long long ll;


int main()
{
    int t;
    cin >> t;
    while(t--)
    {
        int n;
        cin >> n;
        ll sad = 0;
        vector<pair<ll,ll> > vec;
        for(int i = 0; i < n; i++)
        {
            ll a, b;
            cin >> a >> b;
            sad += b;
            vec.push_back(make_pair(a+b, i));
        }
        
        sort(vec.begin(), vec.end());
        ll ans = vec[n-1].first + vec[n-2].first;
        ans -= sad;
        
        cout << ans << endl;
    }
    
}
